me.amankhanna.opendeck.yaml

```diff
@@ -0,0 +1,35 @@
+id: me.amankhanna.opendeck
+
+runtime: org.gnome.Platform
+runtime-version: "47"
+sdk: org.gnome.Sdk
+
+command: opendeck
+finish-args:
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --device=all
+  - --share=ipc
+  - --share=network
+  - --talk-name=org.freedesktop.Flatpak
```

Comment:
this talk name is NOT allowed by default. Why do you need this?

Reply:
OpenDeck supports running optionally plugins compiled for Windows using Wine, and running Node.js plugins using Node. This talk name is used to run the system install of Wine and Node so these optional features don’t have to be bundled with OpenDeck and triple its size as well as cause redundancy if these tools are already installed on the host system.

Reply:
wine is available as a baseapp, that's a large sandbox hole to try using host packages, for some optional features.

Reply:
I have done some superficial research around baseapps and have generally found that it is a sort of template for a Flatpak application, for example an Electron baseapp that provides the libraries and so forth that every Electron app would use. I don't think that a Wine baseapp (if I could even find it) would be the right fit for this, as OpenDeck itself does not run in Wine but instead various plugins.

Various plugins also run their own commands on the host system to interface with user-installed applications and the user also has the capability to specify their own commands to run - it would be very irritating for the user to only be able to run, say, xdg-open because no other commands are available from within the sandbox. I see that this permission was approved for the StreamController application by Core447 for the same reasons, so could the same please be done for OpenDeck?

Reply:
About 50% of them, as OpenDeck is designed to run plugins made for the official Windows/macOS-only official Stream Deck software, and so any plugins written in a compiled language have been compiled only for Windows and macOS. The other 50% are those written with web technologies or those published by myself or other plugin developers who want to support OpenDeck/GNU+Linux better.

Reply:
As for the plugins that run their own commands, my own Starter Pack plugin has a "run command" action used by at least 50 people that I know of for sure that allows the user to specify a command to run on button press and release. For example, interfacing with pipewire to mute and unmute audio devices.

Reply:
https://github.com/ninjadev64/opendeck-starterpack/blob/main/src/run_command.rs

It is yet to be updated to check for the `container` env var to check if it is running in Flatpak and prepend `flatpak-spawn --host` if it is, but you get the idea.

Reply:
For example this Toggle Action with Run Command that one user has:
> `amixer set Capture nocap` and `amixer set Capture cap` for unmute/mute, lol, and it just alternates what command it runs

Reply:
can you open an exception pr for this? https://docs.flathub.org/docs/for-app-authors/linter#exceptions


error code is https://docs.flathub.org/docs/for-app-authors/linter#finish-args-flatpak-spawn-access